### ISSUE TITLE
What is a meta_playground?

### ISSUE THREAD

**Issue by rleonid** [hammerlab/biokepi#174]
Can we play in just a `playground` ?

**Comment by smondet** [comment]
yeah but the word "playground" is already used by Ketrew (for the `Host.t`'s temp-files and so on).

I started with meta-playground a bit as a joke to delay thinking about naming things...

I don't know what to use instead. Any suggestions?

**Comment by ihodes** [comment]
I think "work" directory is reasonable? We're somehow using that for a
subdirectory in the meta_playground, but I don't think we need a separate
name for that subdirectory.

**Comment by rleonid** [comment]
`work_dir` ?

**Comment by smondet** [comment]
The "work" directory you (@ihodes) mention is the `work_dir` argument of the Pipeline compiler:
https://github.com/hammerlab/biokepi/blob/master/src/lib/pipeline.ml#L497

we usually put it inside the the `meta_playground` (to be able to `rm -fr <meta-playground>` :) )

@rleonid so yes  `work_dir` is also already taken :)

**Comment by ihodes** [comment]
Can't the Pipeline compiler use the meta playground? Why does it need its
own directory?

**Comment by smondet** [comment]
@ihodes we could force the compiler's `work_dir` to be a subdir of `meta_playground`.
But since the `meta_playground` otherwise contains more "sample-independent" sutff (software and reference-data); I though it would be handy to separate the 2 (more hackable ways of saving disk-space and/or avoid recomputations for example by sharing among users).

**Comment by ihodes** [comment]
I agree.

@iskander came over and we all chatted; each directory is a work_dir in
their specific contexts; so they can all be called work_dir, and
disambiguated when necessary with prefixes.

**Comment by rleonid** [comment]
@smondet the `work_dir` in the `Pipeline` is used as a prefix of where results are written. AFAIK, we don't actually "work" in that directory. So I think it is misnamed. How about calling that `results_path`? 

The `meta_playgrounds` also don't necessarily have anything executed in them. I think we use them as a destination of where we install tools. How about calling those `install_path` ?

We can go back and change the original `ketrew` playgrounds to actual `work_dir`'s if we really want.

**Comment by smondet** [comment]
@rleonid it's more than the results; the intermediary files go there; and some tools have to `cd` there (like the indel-realigner OTOH).

the `meta_playground` so far has the tools *and* the reference genomes' data (FASTAs + databases like COSMIC).
some tools have to run in there to pre-process the reference genome (like `samtools faidx`, `bwa index`, picard-create-dict) → that preprocessing is reused across samples/runs/experiments.

**Comment by rleonid** [comment]
> the meta_playground so far has the tools and the reference genomes' data (FASTAs + databases like COSMIC).

isn't this more reason to use a more specific name? I find `meta_playground` very uninformative.

**Comment by smondet** [comment]
> isn't this more reason to use a more specific name? I find meta_playground very uninformative.

Yes, me too :)
